src/csvwidget/widget.ts

```diff
@@ -217,6 +217,27 @@ function createDelimiterSwitcherNode(): HTMLElement {
 export
 class CSVWidgetFactory extends ABCWidgetFactory<CSVWidget, DocumentRegistry.IModel> {
   /**
+   * The name of the widget to display in dialogs.
+   */
+  get name(): string {
+    return 'Table';
+  }
+
+  /**
+   * The file extensions the widget can view.
+   */
+  get fileExtensions(): string[] {
+    return ['.csv'];
```

Comment:
The widget factories are registered uniquely by name, so we can't register the same widget factory twice anyway.  Also, we were hiding a lot of details before because the widget factory itself was useless without knowing these values.  I think a reasonable compromise is for the widget factory base class to allow passing these options in its constructor and then have them be read-only.